### PR TITLE
Make torch.serialization.set_default_mmap_options usable as a context manager

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
As title

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #134371

cc @albanD

### CODE REVIEW

**Comment by mikaylagawarecki** [comment]
Adding suppress_bc_linter because this change still allows one to call `torch.serialization.set_default_mmap_options(bool)` as if it were a function

File: torch/serialization.py [hunk 1]
```diff
@@ -54,6 +54,8 @@
     "LoadEndianness",
     "get_default_load_endianness",
     "set_default_load_endianness",
+    "get_default_mmap_options",
+    "set_default_mmap_options",
```
Comment (albanD) [inline torch/serialization.py:58]: Why was this not here before? Did we silence the public API check on it? :o

**Review by albanD — APPROVED** [review]
SGTM as long as public API checks and coverage are green!

File: torch/serialization.py (hunk 1 above)
Comment (mikaylagawarecki) [reply torch/serialization.py:58]: Don't think we silenced, not sure why it wasn't caught

File: test/test_serialization.py [hunk 2]
```diff
@@ -4052,7 +4052,7 @@ def test_serialization_warning_s390x(self):
     @parametrize('path_type', (str, Path))
     @parametrize('weights_only', (True, False))
     @unittest.skipIf(IS_WINDOWS, "NamedTemporaryFile on windows")
-    def test_serialization_mmap_loading(self, weights_only, path_type):
+    def test_serialization_mmap_loading_options(self, weights_only, path_type):
```
Comment (mikaylagawarecki) [inline test/test_serialization.py:4055]:
```suggestion
    def test_serialization_mmap_loading(self, weights_only, path_type):
```

File: test/test_serialization.py [hunk 3]
```diff
@@ -4101,7 +4101,7 @@ def forward(self, input):
             for v in result.values():
                 self.assertTrue(v.is_cuda)
 
-    def test_serialization_mmap_loading_options(self):
+    def test_serialization_mmap_loading(self):
```
Comment (mikaylagawarecki) [inline test/test_serialization.py:4104]:
```suggestion
    def test_serialization_mmap_loading_options(self):
```

**Pull request closed without merging**